### ISSUE TITLE
Test for mixed line ending changes in Travis

### ISSUE THREAD

**Issue by techtonik** [spyder-ide/spyder#2424]
To fix #2335 one of proposed solutions is to automagically convert linefeeds to LF. http://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/ This is bad for two reasons:

1. git may mistake
2. Spyder may lose control over linefeeds and mess them up badly

So, an alternative solution would be to add a Travis check for Windows linefeeds.

1. All files for CRLF (commits to master)
2. Only modified files (http://docs.travis-ci.com/user/pull-requests/)

**Comment by goanpeca** [comment]
@techtonik, the automatic conversion will be implemented, this will be an extra check not an alternative.

**Comment by techtonik** [comment]
@goanpeca, you are free to do what you want. I just haven't seen the argument that could change my own opinion.

**Comment by goanpeca** [comment]
Everyone is entitled to his/her opinion and I am not trying to change yours. I am informing you of what will happen.

**Comment by techtonik** [comment]
The closes I can find on getting the diff http://stackoverflow.com/questions/25071579/list-all-files-changed-in-a-pull-request-in-git-github The remaining problem is to test that it works, and determine current branch for commits to 2.3 etc.

**Comment by techtonik** [comment]
Another awesome concept - https://linthub.io/

**Comment by Nodd** [comment]
1. If git makes mistakes on this, we will have bigger problems. I mean, it's git's job to manage files, it it can't filter by extension or doesn't regognize between CRLF and LF, we're all doomed. Git won't make more mistakes than Travis...
2. This change will impact spyder's own codebase only. In what way would it mess up with other products ?

Now for Travis, my plan would be:
1. Remove all CRLF in all spyder's files (this will be done with `.gitattributes`)
2. Add Travis tests to check all files, don't bother with changes only.

1 and 2 can be added in different PRs independently, @techtonik you seem to know well how to do point 2, could you do it ?

**Comment by ccordoba12** [comment]
@techtonik, there won't be more releases for 2.3, so there's no need to touch that branch :-)

**Comment by techtonik** [comment]
@ccordoba12 I don't agree on this policy. The `master` didn't work for me several days ago. There should always be a stable branch where people can fix bugs and use.

**Comment by techtonik** [comment]
2. @Nodd I suspect that Spyder currently messes with linefeeds and it affects others.

I have a script that checks for linefeeds, but.. Spyder already has linter tools. Can it exposed through command-line?

**Comment by ccordoba12** [comment]
@techtonik, there is a stable branch, and you can move to it by doing

    git checkout 2.3

I maintained it by 9 months, but I simply don't have the time to maintain it forever. So now everything goes to master.

**Comment by Nodd** [comment]
@techtonik Spyder has an option to normalize line endings. You can desactivate it if you find that it messes things up.

AFAIK Spyder's linter tools are external tools like pep8, flake or pylint, they all have a command line interface.

**Comment by techtonik** [comment]
> I maintained it by 9 months, but I simply don't have the time to maintain it forever. So now everything goes to master.

I'd say that there could be a problem with maintenance process and the frequency of releases. From that I've seen in the master, some changes could (should) be applied to 2.3 and then merged back to master.

**Comment by ccordoba12** [comment]
The problem is that we moved to the new (pythonic) style for signals and slots in master (to make Spyder compatible with PyQt5), so that makes really hard to work with both branches.

In any case, there are not major issues with 2.3. If you encounter important problems (other than this one with line endings, which doesn't help to solve any bug), please report them to consider releasing a new bugfix version.

**Comment by techtonik** [comment]
I don't see why new signal slot could not be released as 2.4 if it is still not Python 3 ready and 2.x compatible.

**Comment by techtonik** [comment]
I couldn't find test for mixed linefeeds in existing linter tools. Need to ask explicitly. Please help me and update this post with new information.

* [ ] pep8 (asked - no, prooflink - no)
* [ ] flake (asked - no, prooflink - no)
* [ ] pylint (asked - no, prooflink - no)
* [ ] jedi (asked - no, prooflink - no)

**Comment by Nodd** [comment]
Your link is a 404 error.
How about reading all files in binary and checking if `\r` is in it ? It's quite easy to do with path.py which is included in `spyderlib/utils/exernal`.
I know it's bad to reinvent the wheel but in that case it looks like the wheel doesn't exist...

**Comment by techtonik** [comment]
@Nodd fixed the link. The fact needs to be validated. `pep8` guys say it doesn't belong there. I can argue, but it looks like a there is a technical problem with that.

**Comment by Nodd** [comment]
From your link it exists in pylint:
`pylint a.py -rn --expected-line-ending-format=LF`

**Comment by techtonik** [comment]
Ready for review - https://github.com/spyder-ide/spyder/pull/2470

I feel really bad that so many files are wrong. I'd reimport the stuff into Git again, but I guess there no such option. Who is gonna recommit the files? Should I add them to PR?
